jdaviz/app.py

```diff
@@ -1067,13 +1067,24 @@ def _get_range_subset_bounds(self, subset_state,
                                  simplify_spectral=True, use_display_units=False):
         # TODO: Use global display units
         # units = dc[0].data.coords.spectral_axis.unit
-        viewer = self.get_viewer(self._jdaviz_helper. _default_spectrum_viewer_reference_name)
-        data = viewer.data()
-        if data and len(data) > 0 and isinstance(data[0], Spectrum1D):
-            units = data[0].spectral_axis.unit
+        if self.config == "cubeviz":
```

Comment:
I need to keep the old logic for Cubeviz. Too many things going on there now for me to make sense. But the new logic should be okay for simpler viz (at least didn't break their tests).

Reply:
Kyle, the blame for these lines is a mix of @kecnry (https://github.com/spacetelescope/jdaviz/pull/2127) and @javerbukh (#2087)

Reply:
can you think of any way to merge the logic so there isn't a special case for cubeviz?  Or is there a reason cubeviz has to be handled separately?

Reply:
Cubeviz somehow has `subset_state.att` as a string instead of a proper `att` object that I would expect from Glue. As a result, accessing the `parent` crashes. Do you know why?

Reply:
I'm not sure why, no 🤔 .  Can we just have if-logic to handle that then (get data from `subset_state.att.parent` if possible, otherwise `viewer.data()[0]`)?

Reply:
I guess I could do that, yes.

Reply:
Done.